tests/sentry/api/endpoints/test_monitor_details.py

```diff
@@ -143,81 +205,139 @@ def test_cronjob_crontab(self):
+                resp = self.client.put(
+                    path, data={"config": {"schedule_type": "interval", "schedule": "bar"}}
+                )
 
+                assert resp.status_code == 400, resp.content
+
+    def test_mismatched_org_slugs(self):
+        monitor = self._create_monitor()
+        path = f"/api/0/monitors/asdf/{monitor.guid}/"
+        self.login_as(user=self.user)
+
+        with self.feature({"organizations:monitors": True}):
             resp = self.client.put(
-                self.path, data={"config": {"schedule_type": "interval", "schedule": "bar"}}
+                path, data={"config": {"schedule_type": "interval", "schedule": [1, "month"]}}
             )
 
-            assert resp.status_code == 400, resp.content
+            assert resp.status_code == 400
 
 
+@region_silo_test()
 class DeleteMonitorTest(APITestCase):
-    def test_simple(self):
-        user = self.create_user()
-        org = self.create_organization(owner=user)
-        team = self.create_team(organization=org, members=[user])
-        project = self.create_project(teams=[team])
-
-        monitor = Monitor.objects.create(
-            organization_id=org.id,
-            project_id=project.id,
+    def setUp(self):
+        self.user = self.create_user()
+        self.org = self.create_organization(owner=self.user)
+        self.team = self.create_team(organization=self.org, members=[self.user])
+        self.project = self.create_project(teams=[self.team])
+        self.num_paths = 2
+
+    def _create_monitor(self):
+        return Monitor.objects.create(
+            organization_id=self.org.id,
+            project_id=self.project.id,
             next_checkin=timezone.now() - timedelta(minutes=1),
             type=MonitorType.CRON_JOB,
             config={"schedule": "* * * * *"},
         )
 
-        self.login_as(user=user)
+    def _get_urls(self):
+        return ("sentry-api-0-monitor-details", "sentry-api-0-monitor-details-with-org")
+
+    def _get_path(self, i, monitor):
+        urls = self._get_urls()
+        if i:
+            return reverse(urls[i], args=[self.org.slug, monitor.guid])
+        return reverse(urls[i], args=[monitor.guid])
+
+    def test_simple(self):
+        self.login_as(user=self.user)
         with self.feature({"organizations:monitors": True}):
-            resp = self.client.delete(f"/api/0/monitors/{monitor.guid}/")
+            for i in range(self.num_paths):
+                monitor = self._create_monitor()
+                path = self._get_path(i, monitor)
+
+                resp = self.client.delete(path)
+
+                assert resp.status_code == 202, resp.content
 
-        assert resp.status_code == 202, resp.content
+                monitor = Monitor.objects.get(id=monitor.id)
+                assert monitor.status == MonitorStatus.PENDING_DELETION
+                # ScheduledDeletion only available in control silo
+                assert ScheduledDeletion.objects.filter(
```

Comment:
ScheduledDeletion is a control silo only model so this test cannot be made stable until a service exists for this